cluster_tools/schedulers/slurm.py

```diff
@@ -100,22 +160,61 @@ def inner_submit(
             for resource, value in self.job_resources.items():
                 job_resources_lines += ["#SBATCH --{}={}".format(resource, value)]
 
-        job_array_line = ""
-        if job_count is not None:
-            job_array_line = "#SBATCH --array=0-{}".format(job_count - 1)
-
-        script_lines = (
-            [
-                "#!/bin/sh",
-                "#SBATCH --output={}".format(log_path),
-                '#SBATCH --job-name "{}"'.format(job_name),
-                job_array_line,
-            ]
-            + job_resources_lines
-            + [*additional_setup_lines, "srun {}".format(cmdline)]
+        max_array_size = self.get_max_array_size()
+        max_submit_jobs = self.get_max_submit_jobs()
```

Comment:
I noticed in the tests, that the conig values can also be set to `-1`. I assume, slurm does not return the `-1` as a value if it's set that way (otherwise, the other tests would fail probably), but maybe it's a good idea to ensure `val > 0` in these getters? I could imagine that other slurm versions might yield `-1` if we are unlucky.